martin/src/cog/mod.rs

```diff
@@ -0,0 +1,385 @@
+mod errors;
+
+pub use errors::CogError;
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+use std::vec;
+use std::{fmt::Debug, path::PathBuf};
+
+use std::io::BufWriter;
+use tiff::decoder::{ChunkType, Decoder, DecodingResult};
+use tiff::tags::Tag::{self, GdalNodata};
+
+use async_trait::async_trait;
+use martin_tile_utils::{Format, TileCoord, TileInfo};
+use serde::{Deserialize, Serialize};
+use tilejson::{tilejson, TileJSON};
+use url::Url;
+
+use crate::file_config::FileError;
+use crate::{
+    config::UnrecognizedValues,
+    file_config::{ConfigExtras, FileResult, SourceConfigExtras},
+    MartinResult, Source, TileData, UrlQuery,
+};
+
+#[derive(Clone, Debug)]
+pub struct CogSource {
+    id: String,
+    path: PathBuf,
+    meta: Meta,
+    tilejson: TileJSON,
+    tileinfo: TileInfo,
+}
+
+#[derive(Clone, Debug)]
+struct Meta {
```

Comment:
is it intentional that this is private?

Reply:
Not intentional And pub is better like martin could be used as lib though we haven't got any feedback about this part.

Reply:
That is also possible.
I just thought that maybe having the metadata generation being public might be nice from an API perspective.
On the other hand this would be a bit of work to make nicer like moving `get_meta` to 

```rust
impl Meta {
    pub read(path: &PathBuf) -> Result<Self, FileError> {}
}
```

=> your call

Reply:
Great idea. Let's concentrate on this. Make it separated like martin-mbtiles in the future.